data-prepper-plugins/armeria-common/src/main/java/com/amazon/dataprepper/armeria/authentication/GrpcAuthenticationProvider.java

```diff
@@ -27,4 +31,12 @@ public interface GrpcAuthenticationProvider {
      * @since 1.2
      */
     ServerInterceptor getAuthenticationInterceptor();
+
+    /**
+     * Returns an optional service for HTTP authentication
```

Comment:
We may wish to clarify this. This service is to intercept the HTTP request as part of performing gRPC authentication. Perhaps:

> Allows implementors to provide an {@link HttpService} to either intercept the HTTP request prior to validation, or to perform validation on the HTTP request. This may be optional, in which case it is not used.